x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractor.java

```diff
@@ -207,7 +218,7 @@ Object extractFromSource(Map<String, Object> map) {
                 
                 if (node instanceof List) {
                     List listOfValues = (List) node;
-                    if (listOfValues.size() == 1 || arrayLeniency) {
+                    if ((i < path.length - 1) && (listOfValues.size() == 1 || arrayLeniency)) {
```

Comment:
I would appreciate it if you can add a comment reflecting the reasoning behind this change, in the context of GEO and arrays of values.